src/explore-education-statistics-admin/src/pages/publication/__tests__/PublicationContactPage.test.tsx

```diff
@@ -0,0 +1,251 @@
+    contactTelNo: '0777777777',
+    teamEmail: 'john.smith@test.com',
+    teamName: 'Team Smith',
+  };
+
+  const testPublication: MyPublication = {
+    id: 'publication-1',
+    title: 'Publication 1',
+    contact: testContact,
+    releases: [],
+    legacyReleases: [],
+    methodologies: [],
+    themeId: 'theme-1',
+    topicId: 'topic-1',
+    permissions: {
+      canAdoptMethodologies: true,
+      canCreateReleases: true,
+      canUpdatePublication: true,
+      canUpdatePublicationTitle: true,
+      canUpdatePublicationSupersededBy: true,
+      canCreateMethodologies: true,
+      canManageExternalMethodology: true,
+    },
+  };
+
+  test('renders the contact page correctly', async () => {
+    renderPage(testPublication);
+
+    expect(
+      screen.getByText('Contact for this publication'),
+    ).toBeInTheDocument();
+
+    expect(screen.getByTestId('Team name-key')).toHaveTextContent('Team name');
+    expect(screen.getByTestId('Team name-value')).toHaveTextContent(
+      'Team Smith',
+    );
+
+    expect(screen.getByTestId('Team email-key')).toHaveTextContent(
+      'Team email',
+    );
+    expect(screen.getByTestId('Team email-value')).toHaveTextContent(
+      'john.smith@test.com',
+    );
+
+    expect(screen.getByTestId('Contact name-key')).toHaveTextContent(
+      'Contact name',
+    );
+    expect(screen.getByTestId('Contact name-value')).toHaveTextContent(
+      'John Smith',
+    );
+
+    expect(screen.getByTestId('Contact telephone-key')).toHaveTextContent(
+      'Contact telephone',
+    );
+    expect(screen.getByTestId('Contact telephone-value')).toHaveTextContent(
+      '0777777777',
+    );
+
+    expect(
+      screen.getByRole('button', { name: 'Edit contact details' }),
+    ).toBeInTheDocument();
+  });
+
+  test('clicking the edit button shows the edit form', () => {
+    renderPage(testPublication);
+
+    userEvent.click(
+      screen.getByRole('button', { name: 'Edit contact details' }),
+    );
+
+    expect(screen.getByLabelText('Team name')).toHaveValue('Team Smith');
+    expect(screen.getByLabelText('Team email')).toHaveValue(
+      'john.smith@test.com',
+    );
+    expect(screen.getByLabelText('Contact name')).toHaveValue('John Smith');
+    expect(screen.getByLabelText('Contact telephone')).toHaveValue(
+      '0777777777',
+    );
+    expect(
+      screen.getByRole('button', { name: 'Update contact details' }),
+    ).toBeInTheDocument();
+    expect(screen.getByRole('button', { name: 'Cancel' })).toBeInTheDocument();
+  });
+
+  test('clicking the cancel button switches back to readOnly view', () => {
+    renderPage(testPublication);
+
+    userEvent.click(
+      screen.getByRole('button', { name: 'Edit contact details' }),
+    );
+
+    expect(screen.getByLabelText('Team name')).toHaveValue('Team Smith');
+
+    userEvent.click(screen.getByRole('button', { name: 'Cancel' }));
+
+    expect(screen.getByTestId('Team name-key')).toHaveTextContent('Team name');
```

Comment:
This doesn't seem right as it asserts that the key is itself. This would be better:

```ts
expect(screen.getByTestId('Team name')).toHaveTextContent('Team Smith');
```

We should also to assert other things to ensure that the form is no longer there e.g. the submit button, the fields, etc.

```ts
expect(screen.queryByLabelText('Team name')).not.toBeInTheDocument();
expect(screen.queryByRole('button', { name: 'Update contact details')).not.toBeInTheDocument();
```